### ISSUE TITLE
# Proposal: AI Behavioral Risk Assertions for CycloneDX 2.0

### ISSUE THREAD

**Issue by devashridatta-dotcom** [CycloneDX/specification#956]
This proposal explores a minimal extension to CycloneDX 2.0 that enables machine-readable representation of **AI-specific behavioral risks and exploitability assertions**.

The intent is **not** to introduce a separate exchange format. Rather, the proposal extends existing CycloneDX `risk`, `vulnerability`, and VEX `analysis` constructs to represent AI risks that are not adequately captured by traditional CVEs.

---

## Motivation

CycloneDX already provides:

* Components and services
* Vulnerabilities and CWEs
* VEX analysis
* Risk assessments
* Impact and consequence categories

However, AI systems introduce risks that frequently exist **without a corresponding CVE**.

Examples include:

* Prompt injection
* Indirect prompt injection through retrieved content
* Unsafe tool invocation
* Agent privilege escalation
* Retrieval poisoning
* Unsafe model behavior
* Model behavior drift

Today, these are difficult to represent in a machine-readable and interoperable manner.

---

## Gap Analysis

Current vulnerability models answer:

* Is a CVE applicable?
* Is the component affected?
* Is remediation available?

AI systems often require answering:

* Is the model susceptible to prompt injection?
* Can retrieved content alter model behavior?
* Can the agent invoke tools outside intended boundaries?
* What are the safety consequences if exploitation occurs?

These questions are behavioral and exploitability assertions that may exist independently of a CVE.

---

## Proposed Extension

Introduce AI-specific categories under the existing `risk` object.

Example:

```json
{
  "component": {
    "type": "ml-model",
    "name": "medical-triage-llm"
  },

  "risk": {

    "category": "prompt-injection",

    "analysis": {

      "state": "affected",

      "vector": "indirect",

      "confidence": "high"
    },

    "impact": [

      "health",

      "safety"

    ]
  }
}
```

Possible categories include:

* prompt-injection
* retrieval-poisoning
* unsafe-tool-use
* agent-privilege-escalation
* model-drift
* unsafe-output
* hallucination-risk

---

## Relationship to Existing CycloneDX Constructs

This proposal does not introduce a separate top-level AI security object.

Instead:

* `component`
  represents the model, agent, or AI service

* `risk`
  represents AI behavioral risks

* VEX `analysis`
  represents exploitability and applicability

* `impact`
  represents safety, health, societal, or operational consequences

This approach preserves compatibility with existing CycloneDX 2.0 design principles.

---

## Safety-Critical Example

```json
{
  "component": {
    "type": "ml-model",
    "name": "medical-triage-llm"
  },

  "risk": {

    "category": "prompt-injection",

    "analysis": {

      "state": "affected",

      "vector": "indirect"

    },

    "impact": [

      "health",

      "patient-safety"

    ],

    "rating": "high"

  }
}
```

A consumer or policy engine could then evaluate:

```text
if risk.category == "prompt-injection"
and risk.rating >= HIGH
and analysis.state == affected

then

priority = immediate
```

---

## Discussion

Would CycloneDX benefit from:

1. Standardized AI behavioral risk categories under `risk`?

2. AI-specific exploitability semantics within existing VEX analysis?

3. A common vocabulary for AI behavioral and safety risks that remains compatible with the CycloneDX 2.0 architecture?

I believe this direction preserves existing CycloneDX design principles while enabling machine-readable communication of AI exploitability and behavioral risk assertions.

**Comment by stevespringett** [comment]
Thanks @devashridatta-dotcom. After working this through, categories under `risk` are the wrong home for these values, and the reason is instructive for the whole proposal. Prompt injection is not itself a risk: it carries no likelihood or impact of its own. It is the vehicle by which a threat is realized. The risk is the result: manipulated triage guidance altering patient care, exfiltrated records, unauthorized actions taken by an agent. CycloneDX 2.0 models each layer first class, so your example decomposes rather than flattens:

- The susceptibility is a **weakness**. 2.0 has a weakness object (a CWE identifier or a named weakness), so susceptibility to prompt injection is CWE-1427; retrieval poisoning is a named weakness until CWE coverage catches up.
- The adversarial event is a **threat**, classified by taxonomy (STRIDE, LINDDUN, MAESTRO, MITRE ATT&CK). MAESTRO's layers place agentic AI threats architecturally.
- The evidenced instance is a **vulnerability**, which does not require a CVE. It references the weaknesses that give rise to it, and the evidence work underway in #333 adds a graded exploitability level and labeled supporting data, including prompt and completion transcripts for LLM findings.
- The outcome is the **risk**: impact categories (`safety`, `health`), affected attributes (`integrity`), inherent and residual ratings, linked through `relatedWeaknesses`, `relatedThreats`, and `relatedVulnerabilities`.

Your policy engine query gets stronger, not weaker: if a vulnerability referencing CWE-1427 has exploitability at or above `attacker-reachable` with analysis state `affected`, and a linked risk carries a `safety` impact on a critical asset, then priority is immediate. Every clause is machine checkable against evidence rather than a self applied label.

The genuinely additive piece your list surfaces: MITRE ATLAS is not yet among the threat classification taxonomies, and it is the natural home for AI attack techniques. Adding ATLAS alongside MITRE ATT&CK, with its tactics as the gated categories, is a small change, and I would take your input on it there. The OWASP LLM Top 10 entries are weakness shaped and map through the weakness object rather than a new taxonomy.

**Comment by devashridatta-dotcom** [comment]
Thank you for this,  the decomposition is cleaner and more durable than a flat category approach. Understanding that prompt injection is the vehicle rather than the risk itself, and that the 2.0 architecture already has first-class homes for each layer - weakness, threat, vulnerability, risk - resolves the core framing question.
The policy engine query you sketched is exactly right:
```
if vulnerability references CWE-1427
and exploitability >= attacker-reachable
and analysis.state == affected
and linked risk carries safety impact on critical asset
then priority = immediate
```
Every clause machine-checkable against evidence rather than a self-applied label. This is the governance model AIVEX was trying to reach.

On MITRE ATLAS: yes, adding ATLAS alongside MITRE ATT&CK as a threat classification taxonomy is the right home for AI attack techniques. ATLAS tactics map naturally to the threat layer in the 2.0 architecture and give agentic AI risks a principled classification path rather than a flat category list. I would be glad to contribute input on the ATLAS taxonomy addition. Where is the best place to track that work - a new issue or a comment on an existing one?

On #333 - the evidence model being developed there is exactly the layer AIVEX needs. Graded exploitability and prompt and completion transcripts as labeled supporting data are the right evidence types for AI behavioral risk findings. A prompt injection vulnerability without a reproducible prompt and completion transcript is difficult to triage consistently; with one it becomes machine-checkable against the policy engine query above. 

I would like to follow #333 and contribute to the evidence taxonomy for LLM-specific findings if that is ok with you.

One question on OWASP LLM Top 10: you noted these entries are weakness-shaped and map through the weakness object. Is the expectation that they map to existing CWEs where coverage exists (for example LLM01 Prompt Injection maps to CWE-1427) and to named weaknesses where CWE has not yet caught up, or is there a separate LLM-specific weakness taxonomy being considered?

**Comment by devashridatta-dotcom** [comment]
Following up on the ATLAS taxonomy invitation - I have prepared a field mapping document connecting MITRE ATLAS techniques to CWE, OWASP LLM Top 10, and the CycloneDX 2.0 threat model.
ATLAS to CWE to OWASP LLM Top 10 - core mapping:
ATLAS to CWE mapping — key entries:

LLM Prompt Injection (AML.T0051) — CWE-1427 — LLM01 — Weakness + Vulnerability
Indirect Prompt Injection (AML.T0051.001) — CWE-1427 — LLM01 — Weakness + Vulnerability
Jailbreak (AML.T0054) — CWE-1427 — LLM01 — Threat
Poison Training Data (AML.T0020) — CWE-1039 — LLM03 — Weakness
Backdoor ML Model (AML.T0018) — CWE-506 — LLM03 — Weakness
Steal ML Model (AML.T0030) — CWE-200 — LLM10 — Threat
Invert ML Model (AML.T0024) — CWE-200 — LLM06 — Weakness + Vulnerability
Evade ML Model (AML.T0015) — CWE-1039 — LLM09 — Threat
Compromise Agentic Pipeline (AML.T0052) — No CWE yet — LLM08 — Threat
Manipulate LLM Tool Call (AML.T0056) — No CWE yet — LLM08 — Threat
Autonomous Workflow Chaining (AML.T0058) — No CWE yet — LLM08 — Threat
Five CWE gaps identified:
Three agentic techniques (AML.T0052, AML.T0056, AML.T0058) have no formal CWE yet and are candidates for named weaknesses in the CycloneDX weakness object. Two additional partial mappings: AML.T0024 maps broadly to CWE-200 and AML.T0018 maps broadly to CWE-506, but neither captures the ML-specific mechanism.
Full mapping workbook covers 12 layers:

ATLAS to CWE to OWASP LLM Top 10 (18 techniques)
ML-BOM artifact layer - which AI artifact each technique targets (base weights, chat template, vector DB, tool manifest, fine-tuning adapter, tokenizer, orchestration config)
AIVEX runtime qualifiers - reachability, datasetExposure, chatTemplateUntrusted, toolAccessScope per technique
TMBOM trust boundary graph - 19 directed edges showing exploit propagation across RAG, Training, Inference, Agentic, and Supply Chain pipelines
AI Lifecycle attack mapping - ATLAS techniques per development phase from data collection to retirement
Detection and response guidance per technique with MTTD and severity
Vendor control coverage (NVIDIA NeMo Guardrails, Microsoft Prompt Shields, Google Vertex AI, AWS Bedrock, Anthropic, OpenAI)
Compliance mapping (NIST AI RMF, ISO 42001, EU AI Act, OWASP AI Exchange, NIST SP 800-218, IEC 62304, ISO 26262)

On the OWASP LLM Top 10 question from my previous comment: the mapping confirms LLM01 through LLM10 entries are weakness-shaped and map through the weakness object as you described. Where CWE coverage exists (CWE-1427 for prompt injection) the CWE is the primary reference. Where it does not (agentic techniques), the ATLAS technique ID serves as the named weakness reference pending CWE additions.
On #333: the graded exploitability model and prompt and completion transcripts as evidence types are reflected in the AIVEX runtime qualifier sheet. Happy to contribute to the evidence taxonomy for LLM-specific findings there.
